app/controllers/dragons_teams_controller.rb

```diff
@@ -36,12 +38,40 @@ def destroy
   def add_dragon
     @dragon = current_user.dragons.build(dragon_params)
     if @dragon.save
+      pay_for_dragon
       redirect_to dragons_teams_index_path(current_user.id)
     else
       flash[:alert] = @dragon.errors.full_messages
     end
   end
 
+  def pay_for_dragon
+    resource, user_amount = resources_amount
+    @resource = current_user.resources.find(resource.resource_type.id)
+    @resource.update_attribute(:quantity, user_amount - resource.cost)
+  end
+
+  def can_afford
+    resource, user_amount = resources_amount
+    money?(user_amount, resource.cost)
+  end
+
+  def resources_amount
```

Comment:
Trochę mało opisowa nazwa metody. Mogłaby lepiej objaśniać co się tu dzieje. A jeśli dzieje się za dużo to znak żeby rozbić to na 2 metody.

Dodatkowo proszę zwrócić na to żeby używać bardziej obiektowego podejścia i deklarować zmienne `@` tylko jeśli chce się z nich korzystać w widoku.

Bardziej obiektowo byłoby:
```ruby
dragon_type = DragonType.find(params[:dragons_team][:dragon_type_id])
dragon_cost = dragon_type.dragon_costs[0] # zakładamy że smok może być kupiony za X przedmiotów typu A lub Y przedmiotów typu B itd. ale dla uproszczenia teraz sprawdzamy tylko przedmioty typu A
user_resources_quantity = current_user.resources.find(resource_type: dragon_cost.resource_type).quantity
```

Zgodnie z komentarzem `can_afford` powinno być to przeniesione do modelu User. Chociaż będzie też potrzebne w interaktorze, który zastąpi `pay_for_dragon`.